taskrunner/Services/LotteryCleanupService.cs

```diff
@@ -0,0 +1,114 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using SCJ.Booking.Data;
+using SCJ.Booking.TaskRunner.Utils;
+using Serilog;
+using Task = System.Threading.Tasks.Task;
+
+namespace SCJ.Booking.TaskRunner.Services
+{
+    public class LotteryCleanupService
+    {
+        private readonly ApplicationDbContext _dbContext;
+        private readonly ILogger _logger;
+        private readonly IConfiguration _configuration;
+
+        public LotteryCleanupService(IConfiguration configuration, ApplicationDbContext dbContext)
+        {
+            _dbContext = dbContext;
+            _logger = LogHelper.GetLogger(configuration);
+            _configuration = configuration;
+        }
+
+        /// <summary>
+        ///    Removes names and phone numbers from processed lottery entries
+        /// </summary>
+        public async Task RemovePersonalInfo()
+        {
+            string anonymizedName = "_anonymized";
+            string anonymizedEmail = "_anonymized@example.com";
+
+            _logger.Information("Removing personal info from lottery entries");
+
+            // calculate the oldest lottery entry to keep
+            int maxDays = _configuration.GetValue<int>(
+                "AppSettings:PurgeLotteryContactInfoAfterDays"
+            );
+            DateTime oldestDate = DateTime.UtcNow.AddDays(-maxDays);
+
+            // get entries from ScTrialBookingRequests where ProcessingTimestamp is older than oldestDate
+            var entriesToUpdate = await _dbContext
+                .ScTrialBookingRequests.Where(e => e.ProcessingTimestamp < oldestDate)
+                .Where(e => e.Email != anonymizedEmail && e.RequestedByName != anonymizedName)
+                .ToListAsync();
+
+            _logger.Information(
+                "Updating {count} old entries from ScTrialBookingRequests with anonymized details",
+                entriesToUpdate.Count
+            );
+
+            // Update each entry
+            foreach (var entry in entriesToUpdate)
+            {
+                entry.Email = anonymizedEmail;
+                entry.RequestedByName = anonymizedName;
```

Comment:
also clear`entry.Phone` or set it to an anonymized value

Reply:
If it's just a varchar and not null, I could also set it to the same `anonymizedName` value ("_anonymized") or just an empty string if that feels too weird. Any preference?